### PR TITLE
Update changelog to prepare the next release (0.24)

### PR DESCRIPTION
**Pull request by EmilienM**
Prepare the next release.

### CODE REVIEW

**Comment by EmilienM** [comment]
@mandre @pierreprinetti for review, thanks

**Review by pierreprinetti — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 0.24.0 (December 13, 2021)
+
+IMPROVEMENTS
+
+* Added Get, Update, DeleteKeys, Associate, Disassociate, DisassociateAll, and ListAssociations for blockstorage QoS v3 via [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283) and [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
+* CI: enable Neutron Dynamic Routing service [GH-2271](https://github.com/gophercloud/gophercloud/pull/2271)
```
Comment (mandre) [inline CHANGELOG.md:6]: Does this really belong to the changelog? Looking at the historical content in the changelog, I'd say no.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,12 @@
+## 0.24.0 (December 13, 2021)
+
+IMPROVEMENTS
+
+* Added Get, Update, DeleteKeys, Associate, Disassociate, DisassociateAll, and ListAssociations for blockstorage QoS v3 via [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283) and [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
+* CI: enable Neutron Dynamic Routing service [GH-2271](https://github.com/gophercloud/gophercloud/pull/2271)
+* Bump golang.org/x/crypto [GH-2291](https://github.com/gophercloud/gophercloud/pull/2291)
```
Comment (mandre) [inline CHANGELOG.md:7]: Same here, this could however be listed under `BUG FIX` if this is what justified the bump.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,12 @@
+## 0.24.0 (December 13, 2021)
+
+IMPROVEMENTS
+
+* Added Get, Update, DeleteKeys, Associate, Disassociate, DisassociateAll, and ListAssociations for blockstorage QoS v3 via [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283) and [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
+* CI: enable Neutron Dynamic Routing service [GH-2271](https://github.com/gophercloud/gophercloud/pull/2271)
+* Bump golang.org/x/crypto [GH-2291](https://github.com/gophercloud/gophercloud/pull/2291)
+* Set Go minimum version to 1.14 [GH-2294](https://github.com/gophercloud/gophercloud/pull/2294)
```
Comment (mandre) [inline CHANGELOG.md:8]: Should be listed under `UPGRADE NOTES`, at the top of the changelog.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,12 @@
+## 0.24.0 (December 13, 2021)
+
+IMPROVEMENTS
+
+* Added Get, Update, DeleteKeys, Associate, Disassociate, DisassociateAll, and ListAssociations for blockstorage QoS v3 via [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283) and [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
```
Comment (mandre) [inline CHANGELOG.md:5]: I suggest you list all of the new methods individually, as previously done in past changelog entries.
```suggestion
* Added  `blockstorage/v3/qos.Get` [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283)
* Added  `blockstorage/v3/qos.Update` [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283)
* Added  `blockstorage/v3/qos.DeleteKeys` [GH-2283](https://github.com/gophercloud/gophercloud/pull/2283)
* Added `blockstorage/v3/qos.Associate` [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
* Added `blockstorage/v3/qos.Disassociate` [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
* Added `blockstorage/v3/qos.DisassociateAll` [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
* Added `blockstorage/v3/qos.ListAssociations` [GH-2284](https://github.com/gophercloud/gophercloud/pull/2284)
```

**Review by mandre — COMMENTED** [review]
Let's try to stick to the convention of previous changelog entries.

**Review by mandre — APPROVED** [review]
LGTM :+1:

**Pull request merged**